src/algorithms/graphs/breadthFirstSearch.java

```diff
@@ -0,0 +1,133 @@
+package src.algorithms.graphs;
+
+import java.util.*;
+
+import src.algorithms.graphs.util.BinaryTreeNode;
+import src.algorithms.graphs.util.GraphNode;
+
+/**
+ * Implementation of BFS
+ *
+ * Breadth-First search is a graph traversal algorithm that utilizes a queue (FIFO) data structure
+ * It is useful in finding a shortest-hops solution in graphs
+ * This method is also used to obtain level order traversals of trees.
+ *
+ * In general, BFS works as such:
+ *  - Start with a queue that contains the root node
+ *  - While the queue is not empty:
+ *      - Pop a vertex from the queue
+ *      - Push all neighbours to the queue if they have not been visited yet
+ *      - Update any variables as needed
+ *
+ * Time: O(V + E), where V is the number of vertices/nodes, and E is the number of edges in the graph
```

Comment:
Interestingly, do you know when its strictly (or close to maybe 1-off) to V+E? follow-up: would a complete graph take O(V+E)?